src/java.desktop/share/classes/java/awt/Font.java

```diff
@@ -457,6 +457,12 @@ public FontPeer getFontPeer(final Font font) {
      */
     private transient boolean createdFont = false;
 
+    /*
+     * Font with fallback components (using CompositeFont), its Font2D handle
+     * should be copied to derived fonts.
+     */
+    private transient boolean withFallback;
```

Comment:
I wonder how it will work if the Font was read from the system, then serialized/deserialized and then used?

Reply:
Not sure I fully understand the question. `Font` object cannot have both `createdFont` and `withFallback` fields set, at least in the current implementation.
If we're talking about 'created' fonts (created from a file), the fix doesn't change anything with regard to their behavior - after serialization/deserialization the font might not work, I guess, if it's not registered using `GraphicsEnvironment.registerFont`. In any case, it will lose it's 'created' status, and will change behavior correspondingly.
If we're talking about 'fonts with fallback' (like the ones `JEditorPane` creates under the hood), they will lose their fallback components after serialization/deserialization. But that's also how it worked before the proposed change. I'm not going to address that problem in this PR.